mod: Add 10% damage bonus for 1h swords under strict conditions

Yeldur opened the pull request:
Add 10% damage bonus for 1h swords under strict conditions.

This change applies a 10% melee damage bonus to attacks made with one-handed swords, but only under the following conditions:
- The attack is a swing (not a thrust)
- The damage type is cut
- The agent is not mounted
- The agent has no shield equipped (offhand or in inventory)
- The agent has no ranged weapons in their inventory

The logic is evaluated per attack, ensuring that the bonus dynamically reflects the agent's current state. This incentivizes a reason to play a 1h Sword without a shield.

This has been tested in-game to confirm all of the conditions work as expected.

Video for reference:
https://cdn.discordapp.com/attachments/1186742521255182356/1377032789249101856/1hSwashBucklerBuff2.mp4?ex=68377d38&is=68362bb8&hm=677208b877dc1298df9176f4fd4a626f50c1c025d436058a13edecebd0712d8d&

Yeldur commented:
two issues have been risen ->

1. if the throwable has a pack, even if it is empty then so long as it is in your inventory, the buff is not applied. I'll need to fix this or find a way to address it.

2. it's based on the primary mode of the weapon. Because the primary mode if the Pilum is by default, melee, that means it's considered a melee weapon. This one I do not think I can solve because the Pilum should probably be a throwing weapon first and melee weapon second.

investigating a fix for these.

Muparadzi commented:
Ideally I would like to see this type of bonus mentioned somewhere on the website. If bonuses like this become a norm then we should think about how to make them easily displayed.

Yeldur commented:
ref https://github.com/crpg2/crpg/pull/544/commits/74861ea982cc1aae3e4eb03f09bf23626a06e99b 

Added check to enable damage bonus if throwing pouch is empty (i.e. all throwables in that pouch have been thrown)
Added check to ensure any item with an alternate throwable mode are disqualified from the bonus.

Yeldur commented:
> I think it's okay, I'll make a note to myself: make a feature for 1h swords like this: ![image](https://private-user-images.githubusercontent.com/33551334/449231869-a4cf614c-5c7b-403b-856b-2bbee4936cf2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg2MDk0NDIsIm5iZiI6MTc0ODYwOTE0MiwicGF0aCI6Ii8zMzU1MTMzNC80NDkyMzE4NjktYTRjZjYxNGMtNWM3Yi00MDNiLTg1NmItMmJiZWU0OTM2Y2YyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTMwVDEyNDU0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3OGRkNTcyMzMwNjE5NWEyNDk1Zjk1OGI1NjA1YmRjZTc4MjcwMjRjYmU0MjZhMDQ4NmMyOWE4YjQ4YzgzZTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.F4nLqGXZbl5w6ypljNxfh27jSeRlRY8GhpPgOlx-FDo)

it may be better idea to add something to characteristic page?

if no shield, ranged, horse equipped etc etc

green popup appears stating you get 10% extra damage?

i think feature could confuse some people to believing they get the damage boost all the time
